plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/edit/MySQLConstraintManager.java

```diff
@@ -49,6 +52,56 @@ public DBSObjectCache<MySQLCatalog, MySQLTableConstraint> getObjectsCache(MySQLT
         }
     }
 
+    @Override
+    protected void addObjectCreateActions(
+        @NotNull DBRProgressMonitor monitor,
+        @NotNull DBCExecutionContext executionContext,
+        @NotNull List<DBEPersistAction> actions,
+        @NotNull SQLObjectEditor<MySQLTableConstraint, MySQLTable>.ObjectCreateCommand command,
+        @NotNull Map<String, Object> options
+    ) {
+        if (tryGetColumnOfPrimaryKeyConstraintForAutoincrementColumn(monitor, command.getObject(), false) == null) {
+            super.addObjectCreateActions(monitor, executionContext, actions, command, options);
+        }
+    }
+
+    @Override
+    protected void addObjectDeleteActions(
+        @NotNull DBRProgressMonitor monitor,
+        @NotNull DBCExecutionContext executionContext,
+        @NotNull List<DBEPersistAction> actions,
+        @NotNull SQLObjectEditor<MySQLTableConstraint, MySQLTable>.ObjectDeleteCommand command,
+        @NotNull Map<String, Object> options
+    ) {
+        if (tryGetColumnOfPrimaryKeyConstraintForAutoincrementColumn(monitor, command.getObject(), true) == null) {
+            super.addObjectDeleteActions(monitor, executionContext, actions, command, options);
+        }
+    }
+
+    @Nullable
+    public static MySQLTableColumn tryGetColumnOfPrimaryKeyConstraintForAutoincrementColumn(
+        @NotNull DBRProgressMonitor monitor,
+        @NotNull MySQLTableConstraint tc,
+        boolean checkPersisted
+    ) {
+         if (tc.getConstraintType() == DBSEntityConstraintType.PRIMARY_KEY && checkPersistedState(tc, checkPersisted)) {
+             List<MySQLTableConstraintColumn> tcc = tc.getAttributeReferences(monitor);
+             if (tcc != null && tcc.size() == 1) {
+                 MySQLTableColumn column = tcc.get(0).getAttribute();
+                 if (checkPersistedState(tc, checkPersisted) && column.isAutoGenerated()
+                     && column.getExtraInfo().toLowerCase().contains(MySQLConstants.EXTRA_AUTO_INCREMENT)
+                 ) {
+                     return column;
+                 }
+             }
+         }
+        return null;
+    }
+
+    private static boolean checkPersistedState(DBPPersistedObject persistedObject, boolean checkPersisted) {
+        return checkPersisted && persistedObject.isPersisted() || !checkPersisted && !persistedObject.isPersisted();
```

Comment:
Is this the correct logic? If `checkPersisted` is `true`, we ensure that the object is **persisted**. If it's `false`, we ensure the object is **not persisted**.

If so, then it can be simplified:
```suggestion
        return checkPersisted == persistedObject.isPersisted();
```

Reply:
The logic behind this expression looks less straightforward, but OK